File: Backend/MATCHOP.API/Entities/Court.cs
using MATCHOP.API.Enums;

namespace MATCHOP.API.Entities;

public class Court
{
    public Guid Id { get; set; }


    public Guid VenueId { get; set; }

    public Guid SportId { get; set; }
  
    public string Name { get; set; } = null!;

    public string? Type { get; set; }
    public int? Capacity { get; set; }

    public string? LocationNote { get; set; }

    public string? Description { get; set; }

    public string? ImageUrl { get; set; }


    public string? ImagePublicId { get; set; }

    public CourtStatus Status { get; set; } = CourtStatus.ACTIVE;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public Venue Venue { get; set; } = null!;

    public Sport Sport { get; set; } = null!;

    public ICollection<CourtImage> Images { get; set; } = new List<CourtImage>();

    public ICollection<PriceRule> PriceRules { get; set; } = new List<PriceRule>();

    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();

    public ICollection<BookingSlot> BookingSlots { get; set; } = new List<BookingSlot>();

    public ICollection<CourtBlock> CourtBlocks { get; set; } = new List<CourtBlock>();
}

File: Frontend/src/types/court.ts
export type CourtStatus = "ACTIVE" | "INACTIVE" | "MAINTENANCE";
export type DayType = "ALL" | "WEEKDAY" | "WEEKEND";

export interface Court {
  id: string;
  venueId: string;
  sportId: string;
  name: string;
  type?: string;
  capacity?: number;
  locationNote?: string;
  description?: string;
  imageUrl?: string;
  status: CourtStatus;
  priceRules?: PriceRule[];
  createdAt: string;
  updatedAt: string;
}

export interface PriceRule {
  id: string;
  courtId: string;
  dayType: DayType;
  startTime: string;
  endTime: string;
  pricePerHour: number;
  priority: number;
  status: "ACTIVE" | "INACTIVE";
}

export interface Sport {
  id: string;
  name: string;
  status: "ACTIVE" | "INACTIVE";
}

export interface CourtBlock {
  id: string;
  courtId: string;
  venueId: string;
  startDate: string;
  endDate: string;
  startTime: string;
  endTime: string;
  reason?: string;
  status: "ACTIVE" | "CANCELLED";
}
